pkgs/top-level/perl-packages.nix

```diff
@@ -8016,6 +8016,7 @@ let
       url = "mirror://cpan/authors/id/L/LE/LEONT/File-Map-0.67.tar.gz";
       sha256 = "1hpv4aprgypjxjx1kzbjnf6r29a98rw7mndlinixzk62vyz5sy0j";
     };
+    perlPreHook = "export LD=$CC";
```

Comment:
```suggestion
    perlPreHook = lib.optionalString stdenv.isDarwin "export LD=$CC";
```

Reply:
probably to make it globally for `.isDarwin` and `.is686`, there are too many adhoc fixes

Reply:
For consistency with hundred similar cases among `perl-packages.nix`.
It all of them have `if stdenv.isDarwin` and this one does not, it is to read (by a future refactorer) as "this case is special, unlike the others it is not darwin-only"

Reply:
I do wonder if it is well worth making `export LD=$CC` across all perl module packages that need it (e.g. XS or FFI modules, not pure-Perl modules.)  What we really want is to use whatever `$Config{ld}` was used upon building the perl binary for maximum compatibility with the module being built:


```
zakame:~] 49s % nix-shell -p stdenv -p perl532
[zakame:~] % where perl
/nix/store/192j9y62kp6a5kjw1ylgg1s4w7yxchc5-perl-5.32.1/bin/perl
/home/zakame/.plenv/shims/perl
/usr/bin/perl
[zakame:~] % echo $LD  
ld
[zakame:~] % where $LD     
/nix/store/ca37d3qrydh0wpw40kswsx30j8dyzxh2-gcc-wrapper-10.2.0/bin/ld
/nix/store/jlyaypckqlapwqpck7jscs6qcdwg9bk3-binutils-wrapper-2.35.1/bin/ld
/nix/store/cp1sa3xxvl71cypiinw2c62i5s33chlr-binutils-2.35.1/bin/ld
/usr/bin/ld
[zakame:~] % perl -MConfig -E 'say $Config{ld}'
cc
```

Reply:
Based on the above `$Config{ld}` setup, I'm still leaning towards unconditional `export LD=$CC`.

We should probably revisit refactoring this at the `/development/perl-modules/generic/default.nix` level in another issue for discussion.